backends/ebpf/psa/externs/ebpfPsaMeter.cpp

```diff
@@ -0,0 +1,435 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "ebpfPsaMeter.h"
+#include "backends/ebpf/psa/ebpfPipeline.h"
+
+namespace EBPF {
+
+EBPFMeterPSA::EBPFMeterPSA(const EBPFProgram *program,
+                           cstring instanceName, const IR::Declaration_Instance* di,
+                           CodeGenInspector *codeGen)
+        : EBPFTableBase(program, instanceName, codeGen) {
+    CHECK_NULL(di);
+    auto typeName = di->type->toString();
+    if (typeName == "DirectMeter") {
+        isDirect = true;
+    } else if (typeName.startsWith("Meter")) {
+        isDirect = false;
+        auto ts = di->type->to<IR::Type_Specialized>();
```

Comment:
didn't you make a utility function for this purpose?

Reply:
The `cstring getSpecializedTypeName(...)` does something different. It returns `Type_Specialized` name. Here we take an argument from `Type_Specialized`.